nistats/second_level_model.py

```diff
@@ -0,0 +1,451 @@
+        """ Fit the GLM
+
+        1. create design matrix
+        2. do a masker job: fMRI_data -> Y
+        3. fit regression to (Y, X)
+
+        Parameters
+        ----------
+        first_level_input: list of `FirstLevelModel` objects or pandas
+                           DataFrame or list of Niimg-like objects.
+            If list of `FirstLevelModel` objects, then first_level_conditions
+            must be provided. If a pandas DataFrame, then must contain columns
+            subject_id, map_name, effects_map_path. If list of Niimg-like
+            objects then this will be taken literally as Y for the model fit
+            and design_matrix must be provided.
+
+        first_level_conditions: list of (str or array) or None
+            If first_level_input is a list of `FirstLevelModel` objects then
+            it is mandatory to provide a list of contrast definitions to pass
+            to the compute_contrast method of `FirstLevelModel` which can be
+            a str or array. Check the compute_contrast documentation of
+            `FirstLevelModel` for more details on the contrast definitions.
+
+            If first_level_input is a pandas DataFrame then a list of strings
+            is expected with the names of maps that will be part of the model.
+            None indicates that all maps should be included.
+
+            If first_level_input is a list of Niimg-like objects then this
+            argument will be ignored.
+
+        confounds: pandas DataFrame, optional
+            Must contain a subject_id column. All other columns will be
+            considered as confounders and included in the model. If
+            design_matrix is provided then this argument will be ignored.
+            The resulting second level design matrix will use the same column
+            names as in the given DataFrame for confounds. At least two columns
+            are expected, "subject_id" and at least one confounder.
+
+        design_matrix: pandas DataFrame, optional
+            Design matrix that will be used to fit the GLM. The number of rows
+            in the design matrix must agree with the size of the list in
+            first_level_conditions. Careful that the order of maps given
+            by first_level_conditions or inferred directly from
+            first_level_input when first_level_conditions is None correctly
+            match the order of the rows in the design matrix.
+
+        first_level_conditions_name: list of str or None,
+            This argument only applies when first_level_input is a list of
+            `FirstLevelModel` objects. It specifies the name of the columns
+            corresponding to the provided first_level_conditions.
+            If None then the column names will be generated with the form
+            'con_00'.
+        """
+        # Check parameters
+        # check first level input
+        if isinstance(first_level_input, FirstLevelModel):
```

Comment:
This check is a bit weird. Why not checking that this is indeed a list of FirstleveModels, image or df ?